bioptim/dynamics/configure_problem.py

```diff
@@ -612,9 +747,36 @@ def configure_taudot(nlp, as_states: bool, as_controls: bool):
             If the generalized force derivatives should be a control
         """
 
-        name_tau = [str(i) for i in range(nlp.model.nbGeneralizedTorque())]
+        name_taudot = [str(i) for i in range(nlp.model.nbGeneralizedTorque())]
         ConfigureProblem._adjust_mapping("taudot", ["qdot", "tau"], nlp)
-        ConfigureProblem.configure_new_variable("taudot", name_tau, nlp, as_states, as_controls)
+        ConfigureProblem.configure_new_variable("taudot", name_taudot, nlp, as_states, as_controls)
+
+    @staticmethod
+    def configure_soft_contact_forces(nlp, as_states: bool, as_controls: bool):
+        """
+        Configure the generalized forces derivative
+
+        Parameters
+        ----------
+        nlp: NonLinearProgram
+            A reference to the phase
+        as_states: bool
+            If the generalized force derivatives should be a state
+        as_controls: bool
+            If the generalized force derivatives should be a control
+        """
+        name_soft_contact_forces = []
+        l = ["fx", "fy", "fz"]  # TODO: find a better place to hold this or define it in biorbd ?
+        for ii in range(nlp.model.nbSoftContacts()):
```

Comment:
why `ii`?

Reply:
old habbit from matlab, i is a complex number